crates/swc/tests/tsc-references/unionAndIntersectionInference3_es5.1.normal.js

```diff
@@ -34,17 +34,22 @@ concatMaybe([
     3
 ], 4);
 // Repros from #32247
-var g = _asyncToGenerator(regeneratorRuntime.mark(function _callee(com) {
```

Comment:
This seems like a regression of optimization.

Reply:
Do you mean to add a pure comment?

```
/* #__PURE__ */
```

Reply:
I mean direct call of `_asyncToGenerator` without a wrapper function.

Reply:
This is a trade-off.
If we don't return a new function, the function name will be lost.
[See the Babel playground.](https://babeljs.io/repl#?browsers=chrome%2050&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYewdgzgLgBA5gLhgHgKoBoYCVMGUB8AFKALZKECUMAvPjAJJQCmATgIZQgvK6YbZ0APjACCEAJ5hgjVhy48-mLPiq0YABRYgSASwhM0dajDYSpKfjhgFi28qroz2nbrxiWhos9ObP5bjwokTW09A1QjOgBvACgYGCgACy0AdxhSAG4YgF8MoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.16.6&externalPlugins=&assumptions=%7B%7D)
Maybe we should add a pure comment.

Reply:
And, we need to ensure that the length of this g function is 1 before and after the conversion.

Reply:
Oh, I see. Thank you for the clarification!